src/pages/studyView/StudyViewPageStore.ts

```diff
@@ -3019,13 +3057,42 @@ export class StudyViewPageStore
     ): void {
         trackStudyViewFilterEvent('genomicCategoricalData', this);
 
+        const chart = this._geneSpecificChartMap.get(uniqueKey);
         const dataFilterValues: DataFilterValue[] = getCategoricalFilterValues(
             values
         );
-        this.updateGenomicDataFiltersByValues(
-            uniqueKey,
-            _.cloneDeep(dataFilterValues)
-        );
+        if (chart!.mutationOptionType) {
+            this.updateMutationDataFilters(uniqueKey, [_.cloneDeep(values)]);
+        } else {
+            this.updateGenomicDataFiltersByValues(
+                uniqueKey,
+                _.cloneDeep(dataFilterValues)
+            );
+        }
+    }
+
+    @action.bound
+    updateMutationDataFilters(uniqueKey: string, values: string[][]): void {
+        trackStudyViewFilterEvent('mutationCategoricalData', this);
+
+        if (values.every(valueArray => valueArray.length === 0)) {
+            this._mutationDataFilterSet.delete(uniqueKey);
+        } else {
+            const dataFilterValues: DataFilterValue[][] = values.map(
```

Comment:
maybe a comment explaining why this is necessary

Reply:
Comments added like so:
valueArrays represent a two-dimensional array that supports union and
intersection selection on samples
        
for the first condition, I also added "delete mutationDataFilter if valueArrays is empty"